File: exchange-client/src/models/Currency.ts
import { TimeSeries } from "@/models/TimeSeries";

export interface Currency {
  exchange_rate: number;
  from_symbol: string;
  interval_sec: number;
  last_update_utc: string;
  period: string;
  previous_close: number;
  time_series: { [timestamp: string]: TimeSeries };
  to_symbol: string;
  type: string;
  utc_offset_sec: number;
}

File: exchange-api/src/main/java/com/exchangeapi/exchangeapi/models/currencies/Currency.java
package com.exchangeapi.exchangeapi.models.currencies;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Getter;
import lombok.Setter;

import java.util.Map;
@Setter
@Getter
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class Currency {
    private String fromSymbol;
    private String toSymbol;
    private String type;
    private double exchangeRate;
    private double previousClose;
    private String lastUpdateUtc;
    private Map<String, TimeSeries> timeSeries;
    private int utcOffsetSec;
    private int intervalSec;
    private String period;

}
